### DISCUSSION TITLE
Keep-alive db connection via heartbeat

### DISCUSSION THREAD

**Discussion by mrkeuz** [dbeaver/dbeaver#332] [Q&A] [answered]
Hello,

First of all, I would to thanks for all DBeaver developers and contributors. It's really excellent product! 
I have some proposal.
##### Problem

At now DBeaver losing connection after idle time. So you need reconnect every time. I found [this thread](http://dbeaver.jkiss.org/forum/viewtopic.php?f=2&t=918) about keep-alive problem. And It said:

> Unfortunately this behavior depends on JDBC driver and can't be configured in some generic way. 
##### Proposal

Make connection keep alive via heartbeat queries. A lot of db connection pool's took connection active via empty query like `select 1`, `select * from dual` etc. In some tool you can change this query. I.e. for generic driver.

As an option this can be done via input form in connection window with keep-alive query and heartbeat interval.

Thanks!

**Comment by infectormp** [comment, marked as answer]
DBeaver already have keep-alive settings https://github.com/serge-rider/dbeaver/issues/6

![image](https://cloud.githubusercontent.com/assets/9958813/14231585/e30ee9d4-f991-11e5-8b12-f5093fc25078.png)

**Comment by mrkeuz** [comment]
@infectormp  Thanks! I found. RTFM me :). 
So you can close issue.

**Comment by wangeugene** [comment]
Where is the settings in Dbeaver 7.1 Database -> settings ?

**Comment by jkondratowicz** [comment]
@sail456852 Connection settings -> Initialization
